src/m365/commands/login.ts

```diff
@@ -223,8 +227,26 @@ class LoginCommand extends Command {
       await logger.log(details);
     };
 
-    deactivate();
-    await login();
+
+    try {
```

Comment:
Well yes... We're not doing `deactivate` if the `ensure` option is provided etc. etc... Otherwise `auth.connection.deactivate()` would be called and the user would get loggedout. Meaning it will nog be able to do anything with the cli.
Second, login is called nevertheless because we want the user to be logged in. Think about the case where the access token is expired, then we want to do a silent sign-in again so that the user can stay using the cli. If the user uses ensure and the access token is expired and we don't do anything then the ensure option would be pretty useless i think.

What's your opinion on this @waldekmastykarz , would you like to see things different?

Reply:
I think we've got a few things that aren't inline with the spec.

1. We're always calling `auth.clearConnectionInfo()` which is part of the `logout` command. Why is this necessary. Why are we wiping the whole connection cache while just checking if you're logged in?
2. We're missing comparison of the currently specified authType with the authType on the restored connection. This would be the first check that we have to do to determine if we're ensuring the same type of auth as done previously.
3. If we call the `login` function to ensure that you've got a valid access token, we're basically creating a whole new connection, which is unnecessary. I think we could solve this by calling `await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);` which does the job and has no side effects (recreating connection, activating it, etc).